addOns/retest/src/main/java/org/zaproxy/addon/retest/EditAlertPanel.java

```diff
@@ -0,0 +1,306 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.retest;
+
+import java.awt.CardLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.swing.BorderFactory;
+import javax.swing.DefaultComboBoxModel;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JSpinner;
+import javax.swing.JTextField;
+import javax.swing.ScrollPaneConstants;
+import javax.swing.border.TitledBorder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jdesktop.swingx.JXPanel;
+import org.jdesktop.swingx.ScrollableSizeHint;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.utils.ZapNumberSpinner;
+import org.zaproxy.zap.utils.ZapTextArea;
+import org.zaproxy.zap.utils.ZapTextField;
+import org.zaproxy.zap.view.LayoutHelper;
+
+public class EditAlertPanel extends AbstractPanel {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = LogManager.getLogger(EditAlertPanel.class);
+    private static final Insets DEFAULT_INSETS = new Insets(1, 1, 1, 1);
+
+    private JScrollPane alertPane = null;
+    private ZapTextArea defaultOutput = null;
+    private JXPanel alertDisplay = null;
+    private CardLayout cardLayout = null;
+
+    private ZapNumberSpinner alertEditPlugin = null;
+    private JComboBox<String> alertEditName = null;
+    private ZapTextField alertEditUrl = null;
+    private JComboBox<String> alertEditMethod = null;
+    private ZapTextField alertEditAttack = null;
+    private JComboBox<String> alertEditParam = null;
+    private ZapTextField alertEditEvidence = null;
+    private JComboBox<String> alertEditConfidence = null;
+    private JComboBox<String> alertEditRisk = null;
+    private ZapTextArea alertOtherInfo = null;
+    private DefaultComboBoxModel<String> methodListModel = null;
+    private DefaultComboBoxModel<String> nameListModel = null;
+    private DefaultComboBoxModel<String> paramListModel = null;
```

Comment:
Can methodListModel and paramListModel just be declared where needed, unlike nameListModel they only seem to be used in a single method.